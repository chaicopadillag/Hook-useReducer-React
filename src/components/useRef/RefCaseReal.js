import React, { useState } from 'react';
import MutipleCustomHooks from '../MutipleCustomHooks';

const RefCaseReal = () => {
	const [show, setShow] = useState(false);
	return (
		<div>
			<h2>Use Ref Case Real</h2>
			<hr />
			<button
				className="btn btn-primary mb-2"
				onClick={() => {
					setShow(!show);
				}}>
				{show ? 'Hide' : 'Show'}
			</button>
			{show && <MutipleCustomHooks />}
		</div>
	);
};

export default RefCaseReal;
