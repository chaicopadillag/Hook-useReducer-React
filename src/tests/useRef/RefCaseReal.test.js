import { shallow } from 'enzyme';
import React from 'react';
import RefCaseReal from '../../components/useRef/RefCaseReal';

describe('Prueba en <RefCaseReal/>', () => {
	const wrapper = shallow(<RefCaseReal />);

	test('Debe de Mostrar Corretamente', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('MutipleCustomHooks').exists()).toBe(false);
	});

	test('Debe de Mostrar el componente <MultipleCustomHool/> ', () => {
		wrapper.find('button').simulate('click');
		expect(wrapper.find('MutipleCustomHooks').exists()).toBe(true);
	});
});
