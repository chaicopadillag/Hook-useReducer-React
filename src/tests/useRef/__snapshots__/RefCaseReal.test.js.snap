// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <RefCaseReal/> Debe de Mostrar Corretamente 1`] = `
<div>
  <h2>
    Use Ref Case Real
  </h2>
  <hr />
  <button
    className="btn btn-primary mb-2"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
